frontend/src/components/ManagePermissionsTable/index.vue

```diff
@@ -0,0 +1,632 @@
+            .then((res) => res.data),
+    refetchOnMount: "always",
+    enabled:
+        props.authGroup === "DELEGATED_ADMIN" &&
+        // DelegatedAdminFeatureFlag
+        !environment.isProdEnvironment(),
+    select: (data) => {
+        const sortedByUserName = data.sort((a, b) =>
+            a.user.user_name.localeCompare(b.user.user_name)
+        );
+        // Move matching IDs to the start of the array
+        return [
+            ...sortedByUserName.filter((item) =>
+                newDelegatedAdminIds.includes(item.access_control_privilege_id)
+            ),
+            ...sortedByUserName.filter(
+                (item) =>
+                    !newDelegatedAdminIds.includes(
+                        item.access_control_privilege_id
+                    )
+            ),
+        ];
+    },
+});
+
+const tableFilter = ref({
+    global: { value: "", matchMode: FilterMatchMode.CONTAINS },
+});
+
+const handleSearchChange = (newValue: string) => {
+    tableFilter.value.global.value = newValue;
+};
+
+const getTableRows = () => {
+    switch (props.authGroup) {
+        case "FAM_ADMIN":
+            return appAdminQuery.data.value ?? [];
+        case "APP_ADMIN":
+            return appUserQuery.data.value ?? [];
+        case "DELEGATED_ADMIN":
+            return delegatedAdminQuery.data.value ?? [];
+        default:
+            return [];
+    }
+};
+
+// Get the query loading status
+const isQueryLoading = (): boolean => {
+    switch (props.authGroup) {
+        case "FAM_ADMIN":
+            return appAdminQuery.isLoading.value;
+        case "APP_ADMIN":
+            return appUserQuery.isLoading.value;
+        case "DELEGATED_ADMIN":
+            return delegatedAdminQuery.isLoading.value;
+        default:
+            return false;
+    }
+};
+
+// Get the query fetching status
+const isQueryError = (): boolean => {
+    switch (props.authGroup) {
+        case "FAM_ADMIN":
+            return appAdminQuery.isError.value;
+        case "APP_ADMIN":
+            return appUserQuery.isError.value;
+        case "DELEGATED_ADMIN":
+            return delegatedAdminQuery.isError.value;
+        default:
+            return false;
+    }
+};
+
+const getQueryErrorValue = () => {
+    let error = null;
+    switch (props.authGroup) {
+        case "FAM_ADMIN":
+            error = appAdminQuery.error.value;
+            break;
+        case "APP_ADMIN":
+            error = appUserQuery.error.value;
+            break;
+        case "DELEGATED_ADMIN":
+            error = delegatedAdminQuery.error.value;
+            break;
+    }
+    if (error) {
+        return isAxiosError(error)
+            ? `Failed to fetch data. ${formatAxiosError(error)}`
+            : "Failed to fetch data.";
+    }
+
+    return undefined;
+};
+
+const handleAddButton = () => {
+    if (props.authGroup === "FAM_ADMIN") {
+        router.push({ name: AddFamPermissionRoute.name });
```

Comment:
Using `authGroup` to do switching is really confusing.
If I am a FAM Admin, then I can add Application Admin but here says "AddFamPermissionRoute" route.

If I am an Application Admin, then I can add "Delegated Admin" permission or "User" permission, but it says "AddAppPermissionRoute.name".

Maybe with some comment would help.

Reply:
I would argue that whatever we had before was really confusing. 
If you are a FAM admin, you are adding FAM permissions for other people to manage their apps on FAM, so you're adding FAM permissions, if they don't have permissions, we show them that they don't have any permissions _in FAM_. 

If you are a delegated admin or app admin, then you are grating app permissions.

Reply:
In short:
FAM Admin -> Add fam permissions
App Admin and Delegated Admin -> Add app permissions

Reply:
could you elaborate further on why authGroup is confusing? it's used to determine what type table to be shown. 
If you are a FAM admin you get a FAM admin table, clicking on the button lets you add FAM permissions.
If you are an App Admin you get an App admin table
If you are a D-Admin, you get a D-admin table

Reply:
We had convention to call each different admin level as "FAM_ADMIN", "APP_ADMIN" and "DELEGATED_ADMIN" but that's from the auth privileges point of view. We used to call _adding application permission for user_ as referring to adding "regular" user permission.

It is just a bit confusing when reading "FAM Admin -> Add fam permissions" because FAM app itself is about adding permissions (application admin permission, delegated admin permission, regular user permission), and "fam permissions" kind of is a general meaning for me so I was confused.

You probably mean "FAM Admin -> can Add application admin permission".

Reply:
https://github.com/bcgov/nr-forests-access-management/pull/1650#issuecomment-2477887224

i did the changes here, do you think this addresses the issue?

Reply:
Suggest suffix with "table" might be clearer, but yours is good enough if you feel redundant.
![image](https://github.com/user-attachments/assets/82cb48d3-2f66-47a0-8b56-e384e72c7981)

Reply:
![image](https://github.com/user-attachments/assets/82590e58-4ac0-45bd-8143-f8ee55e32618)
https://github.com/bcgov/nr-forests-access-management/pull/1650/commits/1647b0ba31143fab43f93b7f79a44741925554ee

I changed it to a Enum, so we have to do ManagePermissionsTableEnum.AppUser

I think the name User Table can be confusing, since the _user_ term is too general.

FamAppAdmin -> A group of users that are admin under an app
AppUser -> A group of users under an app
DelegatedAdmin -> A group of d-admin under an app 

what do you think of this

Reply:
Maybe for the "FamAppAdmin" just use "AppAdmin".
If you don't prefer enum, 'const' does also work right? Just use your type original and substitute values with const values I guess will work. 
I don't mean to use `enum` all the way, previously I just want to make sure we don't need to hardcode values and still can get the benefit of typing.

Reply:
Sure, and no i think it's good to use enum here

Reply:
I liked FamAppAdmin because it corresponded well with the `FamAppAdminGetResponse` type

updated here https://github.com/bcgov/nr-forests-access-management/pull/1650/commits/e0ea63e8cef19c79f3c891f32b49b6c703099f1d

Reply:
I resolved the other conversation, let's chat here

Reply:
Ah~~ I see the reason why now, because you related it to `**Fam**AppAdminGetResponse`. It is a schema object. In fact, there are many schema/model/function objects in backend prefix with unnecessary "FAM_" for everything. I think we had headache with that previously in some conversation.
It's ok if you like to keep it, I don't insist it :) (I thought without that was clearer to me, but maybe it is clearer to others with it).